### PR TITLE
Make AnkiDroid night mode follow system night mode

### PR DESCRIPTION
**Pull request by agentydragon**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
_Link to the issues._

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code

### CODE REVIEW

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4760239&bid=15873458)

File: AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.java [hunk 1]
```diff
@@ -86,21 +95,14 @@ protected void initNavigationDrawer(View mainView) {
             // Decide which action to take when the navigation button is tapped.
             toolbar.setNavigationOnClickListener(v -> onNavigationPressed());
         }
+        mSystemNightMode = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
         // Configure night-mode switch
-        final SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(NavigationDrawerActivity.this);
+        final SharedPreferences preferences = getPreferences();
+        ;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.java]: Codacy Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=4760239)

**Comment by mikehardy** [comment]
Not commenting on the approach or when to apply discussion (it's ongoing and I'm just listening) but code-wise this is the way to go about it based on my experience implementing similar in react-native. Here's the implementation module most are using there if it's interesting to you for ideas:

https://github.com/expo/react-native-appearance/blob/master/android/src/main/java/io/expo/appearance/RNCAppearanceModule.java

**Comment by mikehardy** [comment]
With #5692 you might want to rebase this now and re-push, and I can review at long last?

**Comment by agentydragon** [comment]
Sorry for some days of radio silence. I'm not sure when I'll have more time to give to this to finish it. I'm not sure it's ready to be merged as it is - if I remember correctly, this actually removes the option to control dark mode manually, which some users might want to keep.

**Comment by mikehardy** [comment]
@agentydragon no worries, it's all a bit crazy right now and all/any delay is acceptable of course. My read on it (without testing - which I would do prior anyway) is that it allows user manual preferences to both exist and override, and I do believe this is desired - if a user wants manual control I want to allow that. I tagged it for 2.10 which won't go for another month or two and if you don't have time between now and then I can locally pull it and make the fixes needed to carry it across the line. Cheers!

**Comment by mikehardy** [comment]
Just installed the (relatively) new github app and noticed they just offered "day" / "night" / "follow system" as the three preferences. Seemed clean and tidy and left the user in control

**Comment by david-allison** [comment]
Superseded by #10520

**Pull request closed without merging**